src/v/cluster/log_eviction_stm.cc

```diff
@@ -40,82 +41,123 @@ log_eviction_stm::log_eviction_stm(
   ss::abort_source& as,
   storage::kvstore& kvstore)
   : persisted_stm("log_eviction_stm.snapshot", logger, raft, kvstore)
-  , _logger(logger)
-  , _as(as) {}
+  , _as(as)
+  , _queue(max_event_queue_size) {}
 
 ss::future<> log_eviction_stm::start() {
     ssx::spawn_with_gate(_gate, [this] { return monitor_log_eviction(); });
     ssx::spawn_with_gate(
-      _gate, [this] { return write_raft_snapshots_in_background(); });
+      _gate, [this] { return handle_log_eviction_events(); });
     return persisted_stm::start();
 }
 
 ss::future<> log_eviction_stm::stop() {
-    _reap_condition.broken();
+    _queue_mutex.broken();
```

Comment:
I think we need to `_as.request_abort()` and close the gate. Otherwise the event loop may use after aborting?

Reply:
This is pretty confusing but the abort source passed to all of the stms is managed externally, in this case by `cluster::partition`, it called `_as.request_abort()` in `cluster::partition::stop()`